### PR TITLE
Configure sessions

### PR DESCRIPTION
**Pull request by JordT**
We can authenticate a user... 🚀

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,3 +12,15 @@ Objectives
 
 
 Project will be a login Page allowing admins to increment a counter and users to view the count.
+
+https://medium.com/@ethantcollins98/setting-and-using-cookies-with-a-node-js-express-server-49479673d043
```
Comment (simonireilly) [inline README.md:16]: :hot_pepper:

File: src/components/Login.js [hunk 2]
```diff
@@ -31,14 +29,27 @@ const Login = () => {
...
-                (res) => setResponse('ERROR: User not authenticated.'))
+            .post('http://localhost:4000/auth', auth, {withCredentials: true})
+            .then((x) => {
+                console.log(x);
+                console.log('Routing to /session from React')
+                // window.location.href="/session"
+            })
+    }
+
+    const authTest = (auth) => {
```
Comment (simonireilly) [inline src/components/Login.js:40]: Consider abstraction

```
const API_URL = "http://localhost:4000"
```

When you deploy this to staging/prod you will need to configure server URL

File: README.md [hunk 3]
```diff
@@ -12,3 +12,17 @@ Objectives
 
 
 Project will be a login Page allowing admins to increment a counter and users to view the count.
+
```
Comment (simonireilly) [inline README.md:15]: Brief description, why should I read these, so maintainer knows why they are there

File: src/server/server.js [hunk 4]
```diff
@@ -32,42 +37,25 @@ mongoose
...
-  next();});
+app.use(cors({
+  origin: 'http://localhost:3000', 
+  credentials: true
+}))
 app.use(express.urlencoded({extended: true,}));
 app.use(express.json());
-app.use(session({secret: "secret"})); // Set up express-sessions
+app.use(session({ //setup session middleware
+  secret: "secret",
```
Comment (simonireilly) [inline src/server/server.js:47]: Guess this has to be a cyrto secure string.

Consider a 32 length random hexadecimal.

Store in an env var e.g. process.env.SESSION_SIGNING_SECRET

File: src/server/server.js [hunk 5]
```diff
@@ -32,42 +37,25 @@ mongoose
...
-    "GET, POST, OPTIONS, PUT, PATCH, DELETE"
-  );
-  res.setHeader(
-    "Access-Control-Allow-Headers",
-    "X-Requested-With,content-type"
-  );
-  res.setHeader("Access-Control-Allow-Credentials", true);
-  next();});
+app.use(cors({
+  origin: 'http://localhost:3000',
```
Comment (simonireilly) [inline src/server/server.js:41]: Abstract URL to const so we can configure at deploy time

File: src/server/server.js [hunk 6]
```diff
@@ -32,42 +37,25 @@ mongoose
   )
   .catch((err) => log(serverLog(err)));
 
-// **** middleware ****//
```
Comment (simonireilly) [inline src/server/server.js:35]: Tell me why we deleted this?

**Review by simonireilly — COMMENTED** [review]
She's a beauty, some comments but happy to merge, just let me know playaaaaa

File: src/server/server.js (hunk 6 above)
Comment (JordT) [reply src/server/server.js:35]: I was waging a war against the green text 👊

File: src/server/server.js (hunk 4 above)
Comment (JordT) [reply src/server/server.js:47]: Will pop this in an issue for future us 👍

File: README.md (hunk 3 above)
Comment (JordT) [reply README.md:15]: This is my dump of links that might save me.... 👎

**Pull request merged**